### PR TITLE
Craft Arms & Armor Update

### PR DESCRIPTION
**Pull request by doug1234**
Added new column for fixed extra price to arms and armor.  Improved crafted items long descriptions.  Added energy resistance armor and shields.

### CODE REVIEW

File: TemplePlus/ui/ui_item_creation.cpp [hunk 1]
```diff
@@ -587,6 +589,10 @@ bool UiItemCreation::IsOutmoded(int effIdx){
 		for (auto it: appliedBonusIndices){
 			if (it == itEnh->upgradesTo)
 				return true;
+
+			//Removes the start effect after it is added (otherwise it displays the greyed start effect and the second effect)
+			if ((it == effIdx) && (itEnh->downgradesTo == CRAFT_EFFECT_INVALID))
```
Comment (DudeMcDude) [inline TemplePlus/ui/ui_item_creation.cpp:594]: Is this well tested?

File: TemplePlus/ui/ui_popup.cpp [hunk 2]
```diff
@@ -3,27 +3,26 @@
 #include "ui.h"
 #include "tig/tig_keyboard.h"
 
-UiPopup uiPopup;
+UiPopupHandler uiPopupHandler;
```
Comment (DudeMcDude) [inline TemplePlus/ui/ui_popup.cpp:6]: I think I've unified this with UiPopup in my branch

File: TemplePlus/ui/ui_item_creation.cpp (hunk 1 above)
Comment (doug1234) [reply TemplePlus/ui/ui_item_creation.cpp:594]: Define "well tested"...  it was definitely displaying strange as it was.  I made the change and it no longer displayed the greyed out first effect for spell resistance armor and the new energy armor.  I tested adding and removing effects and all looked good.  Is there anything else I should have tried?

File: TemplePlus/ui/ui_popup.cpp (hunk 2 above)
Comment (doug1234) [reply TemplePlus/ui/ui_popup.cpp:6]: Right.  I changed a few things around in addition to renaming.  I see a messy merge in my future...

File: TemplePlus/ui/ui_item_creation.cpp (hunk 1 above)
Comment (DudeMcDude) [reply TemplePlus/ui/ui_item_creation.cpp:594]: Does it just affect display? What happens if you take an already crafted item and then remove an effect?
I just have a vague feeling of deja vu here...

Comment (doug1234) [reply TemplePlus/ui/ui_item_creation.cpp:594]: It is display only but odd enough I thought I had added a bug at first.  I am pretty sure I tested adding then backing out effects and everything was good.  I'll give it another test after my merge though.

Comment (DudeMcDude) [reply TemplePlus/ui/ui_item_creation.cpp:594]: Just to be clear, I'm talking about the case where you hit the craft button and reopen the craft UI to add more stuff. You need to make sure that doesn't go haywire.

Comment (doug1234) [reply TemplePlus/ui/ui_item_creation.cpp:594]: I can add an effect, craft the item and then come back, add the next effect and craft again.  If that is what you mean then it works.  I have seen another graphical oddity where effects show up double on the right side but that is a different (display only) issue I'll look into at another time.  Merged and crafting still seems to be working correctly.

**Pull request merged**